src/routes/users.js

```diff
@@ -1,271 +1,324 @@
+    });
     return;
   }
   try {
     const token = await auth0_client.get_token();
     const auth0_user = await auth0_client.get_user_by_id(user_id, token);
-    if (!auth0_user){
-      res.status(404).send({user: null, message: "User is not registered in Auth0"});
+    if (!auth0_user) {
+      res
+        .status(404)
+        .send({ user: null, message: "User is not registered in Auth0" });
       return;
     }
-    let db_user = await prisma.users.findUnique({ 
+    let db_user = await prisma.users.findUnique({
       where: { id: user_id },
       include: {
         major_dept: true,
         d_major_dept: true,
-      }
+      },
     });
-    if(db_user){
+    if (db_user) {
       db_user = await process_user_info(db_user);
-      res.status(200).send({user: {db: db_user, auth0: auth0_user}, message: "Successfully get user by id."});
-    }else{
-      res.status(400).send({user: null, message: "User not found in DB."});
+      res.status(200).send({
+        user: { db: db_user, auth0: auth0_user },
+        message: "Successfully get user by id.",
+      });
+    } else {
+      res.status(200).send({ user: null, message: "User not found in DB." });
     }
-  }
-  catch (err) {
-    res.status(500).send({user: null, message: err});
+  } catch (err) {
+    res.status(500).send({ user: null, message: err });
     const fields = [
-      {name: "Component", value: "Backend API endpoint"},
-      {name: "Method", value: "GET"},
-      {name: "Route", value: "/users/:id"},
-      {name: "Request Body", value: "```\n"+JSON.stringify(req.body)+"\n```"},
-      {name: "Error Log", value: "```\n" + err + "\n```"}
-    ]
-    await sendWebhookMessage("error","Error occurred in ncn-backend.", fields);
+      { name: "Component", value: "Backend API endpoint" },
+      { name: "Method", value: "GET" },
+      { name: "Route", value: "/users/:id" },
+      {
+        name: "Request Body",
+        value: "```\n" + JSON.stringify(req.body) + "\n```",
+      },
+      { name: "Error Log", value: "```\n" + err + "\n```" },
+    ];
+    await sendWebhookMessage("error", "Error occurred in ncn-backend.", fields);
     console.error(err);
   }
-})
+});
 
 // API version: 2.0
-router.post('/', checkJwt, async (req, res) => {
+router.post("/", checkJwt, async (req, res) => {
   const email = req.body.user.email;
   const token_sub = req.user.sub;
-  try{
-    if(!email){
-      res.status(400).send({message: "email is required", user: null});
+  try {
+    if (!email) {
+      res.status(400).send({ message: "email is required", user: null });
       return;
     }
-    const db_user = await prisma.users.findUnique({ where: { email: email } });
-    if(db_user){
-      res.status(400).send({message: "email is already registered", user: null});
+    const db_user = await prisma.users.findFirst({ where: { email: email } }); // workaround here bcz email is not unique, so can't use findUnique
```

Comment:
Bug 4: When registering, using `findUnique` will invoke error in backend (you can try by deleting your account and login again and observe error in backend). I use `findFirst` as workaround, it works fine now.